File: frontend/src/constants/shipmentStatus.ts
import { ShipmentStatus } from '@/types/shipment';
import { Tone } from '@/types/ui';

export const SHIPMENT_STATUS_LABEL: Record<ShipmentStatus, string> = {
  LOADED: '배차대기',
  DISPATCHED: '배차완료',
  IN_DELIVERY: '배송중',
  DELIVERED: '배송완료',
  FAILED: '배송실패',
};

export const SHIPMENT_STATUS_TONE: Record<ShipmentStatus, Tone> = {
  LOADED: 'gray',
  DISPATCHED: 'blue',
  IN_DELIVERY: 'green',
  DELIVERED: 'green',
  FAILED: 'red',
};

File: backend/src/main/java/com/logistics/domain/loading/domain/ShipmentStatus.java
package com.logistics.domain.loading.domain;

/** 상차완료 → 배차완료 → 배송중 → 배송완료 | 배송실패 */
public enum ShipmentStatus {
    LOADED,
    DISPATCHED,
    IN_DELIVERY,
    DELIVERED,
    FAILED,
}
